### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Add guide to list all breaking changes and new features

This guide will be updated during the developement of Kuzzle 2.x

### CODE REVIEW

File: doc/2/guides/migrate-from-v1/introduction/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - End of Node.js 6 support
+  - End of Redis 3 and 4 support
+  - End of Elasticsearch 5 support
+
+## New Features
+
+  - Support for Node.js 10
+  - Support for Elasticsearch 7
+
+## Enhancements
```
Comment (alexandrebouthinon) [inline doc/2/guides/migrate-from-v1/introduction/index.md]: If there is no enhancements why keeping this part?
```suggestion
```

Comment (Aschen) [reply doc/2/guides/migrate-from-v1/introduction/index.md]: There may be enhancements later

Comment (scottinet) [reply doc/2/guides/migrate-from-v1/introduction/index.md]: Shouldn't we just list breaking changes, and let the changelog generator take care of the rest when we release?

File: doc/2/guides/migrate-from-v1/introduction/index.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+type: branch
+title: Migration Guide
+order: 100
+---
+
+# Migrate from Kuzzle v1 to Kuzzle v2
+
+## Breaking changes
+
+  - End of Node.js 6 support
```
Comment (scottinet) [inline doc/2/guides/migrate-from-v1/introduction/index.md]:
```suggestion
  - End of Node.js 6 and 8 support
```

**Pull request merged**